### DISCUSSION TITLE
Delete all tables from a specific schema version

### DISCUSSION THREAD

**Discussion by Sandrichka22** [simolus3/drift#2861] [Q&A] [answered]
Is there a way to delete all the tables from a specific schema version (not being the latest one) ?
I am asking because I have something like the following (I'm using the migrator function):
```
from55To56: ...
from56To57: (m, schema) async {
        for (final table in allTables) {
          await m.deleteTable(table.actualTableName);
          await m.createTable(table);
        }
        await m.recreateAllViews();
      }, 
from57To58: (m, schema) async {
        await m.addColumn(playerEntity, playerEntity.playerType);
        await m.recreateAllViews();
      }
```
where in version 57 all tables were deleted and re-created again, and now in the newest version 58 I would like to add a new column to a table. Using the `allTables` getter to delete all the tables does not work anymore, as it now refers to the latest schema which has a new column.
Thanks in advance for the help!

**Comment by simolus3** [comment, marked as answer]
> Using the `allTables` getter to delete all the tables does not work anymore

You have access to the schema at a specific version through the `schema` callback passed to these migrations.

So you can use

```dart
from56To57: (m, schema) async {
        for (final table in schema.entities.whereType<TableInfo>()) {
          await m.deleteTable(table.actualTableName);
          await m.createTable(table);
        }
        await m.recreateAllViews();
      }, 
```

**Reply by Sandrichka22** [reply]
Got it! Thanks so much for your quick reply
